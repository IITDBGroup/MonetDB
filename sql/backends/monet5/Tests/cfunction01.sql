-- continues functions
create stream table ftmp(i integer) set window 2;

create function aggr01()     
returns integer
begin
    declare s int;
	set s = 0;
	while (true)
	do
		set s = s + (select count(*) from ftmp);
		yield s ; 
	end while;
	return s;
END;
select * from functions where name ='aggr01';

-- a continuous function can be called used like any other function?
select aggr01(); #should return 0

start continuous function aggr01();
call cquery.wait(1000); #wait to be started

select aggr01(); #should return 0
pause continuous aggr01;

insert into ftmp values(1),(1);
resume continuous aggr01;

call cquery.wait(1000); #wait for processing
select aggr01(); #should return 2

pause continuous aggr01;
insert into ftmp values(2),(2);
insert into ftmp values(3),(3);

resume continuous aggr01;
call cquery.wait(1000);
select aggr01(); #should return 6

call cquery.wait(1000);
select aggr01(); #should return 6

stop continuous aggr01;
drop function aggr01;
drop table ftmp;
