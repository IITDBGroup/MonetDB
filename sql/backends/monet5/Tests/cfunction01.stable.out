stdout of test 'cfunction01` in directory 'sql/backends/monet5` itself:


# 16:27:02 >  
# 16:27:02 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=36085" "--set" "mapi_usock=/var/tmp/mtest-2361/.s.monetdb.36085" "--set" "monet_prompt=" "--forcemito" "--dbpath=/home/ferreira/MonetDB-trails/BUILD/var/MonetDB/mTests_sql_backends_monet5"
# 16:27:02 >  

# MonetDB 5 server v11.28.0
# This is an unreleased version
# Serving database 'mTests_sql_backends_monet5', using 8 threads
# Compiled for x86_64-pc-linux-gnu/64bit with 128bit integers
# Found 15.498 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2017 MonetDB B.V., all rights reserved
# Visit https://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://dhcp-120.eduroam.cwi.nl:36085/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-2361/.s.monetdb.36085
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded
# MonetDB/Timetrails module loaded

Ready.
# SQL catalog created, loading sql scripts once
# loading sql script: 09_like.sql
# loading sql script: 10_math.sql
# loading sql script: 11_times.sql
# loading sql script: 12_url.sql
# loading sql script: 13_date.sql
# loading sql script: 14_inet.sql
# loading sql script: 15_querylog.sql
# loading sql script: 16_tracelog.sql
# loading sql script: 17_temporal.sql
# loading sql script: 18_index.sql
# loading sql script: 20_vacuum.sql
# loading sql script: 21_dependency_functions.sql
# loading sql script: 22_clients.sql
# loading sql script: 23_skyserver.sql
# loading sql script: 25_debug.sql
# loading sql script: 26_sysmon.sql
# loading sql script: 27_rejects.sql
# loading sql script: 39_analytics.sql
# loading sql script: 39_analytics_hge.sql
# loading sql script: 40_geom.sql
# loading sql script: 40_json.sql
# loading sql script: 40_json_hge.sql
# loading sql script: 41_md5sum.sql
# loading sql script: 45_uuid.sql
# loading sql script: 46_profiler.sql
# loading sql script: 50_cquery.sql
# loading sql script: 51_sys_schema_extension.sql
# loading sql script: 60_wlcr.sql
# loading sql script: 72_fits.sql
# loading sql script: 74_netcdf.sql
# loading sql script: 75_lidar.sql
# loading sql script: 75_shp.sql
# loading sql script: 75_storagemodel.sql
# loading sql script: 80_statistics.sql
# loading sql script: 80_udf.sql
# loading sql script: 80_udf_hge.sql
# loading sql script: 85_bam.sql
# loading sql script: 90_generator.sql
# loading sql script: 90_generator_hge.sql
# loading sql script: 99_system.sql

# 16:27:02 >  
# 16:27:02 >  "mclient" "-lsql" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-2361" "--port=36085"
# 16:27:02 >  

#create stream table ftmp(i integer) set window 2;
#create function aggr01()     
#returns integer
#begin
#    declare s int;
#	set s = 0;
#	while (true)
#	do
#		set s = s + (select count(*) from ftmp);
#		yield s ; 
#	end while;
#	return s;
#END;
#select * from functions where name ='aggr01';
% sys.functions,	sys.functions,	sys.functions,	sys.functions,	sys.functions,	sys.functions,	sys.functions,	sys.functions,	sys.functions,	sys.functions # table_name
% id,	name,	func,	mod,	language,	type,	side_effect,	varres,	vararg,	schema_id # name
% int,	varchar,	varchar,	varchar,	int,	int,	boolean,	boolean,	boolean,	int # type
% 4,	6,	175,	4,	1,	1,	5,	5,	5,	4 # length
[ 8669,	"aggr01",	"create function aggr01() \nreturns integer\nbegin\n declare s int;\n set s = 0;\n while (true)\n do\n set s = s + (select count(*) from ftmp);\n yield s ; \n end while;\n return s;\nend;",	"user",	2,	1,	false,	false,	false,	2000	]
#select aggr01(); #should return 0
% .L2 # table_name
% L2 # name
% int # type
% 1 # length
[ 0	]
#start continuous function aggr01();
#select aggr01(); #should return 0
% .L2 # table_name
% L2 # name
% int # type
% 1 # length
[ 0	]
#pause continuous aggr01;
#insert into ftmp values(1),(1);
[ 2	]
#resume continuous aggr01;
#select aggr01(); #should return 2
% .L2 # table_name
% L2 # name
% int # type
% 1 # length
[ 2	]
#pause continuous aggr01;
#insert into ftmp values(2),(2);
[ 2	]
#insert into ftmp values(3),(3);
[ 2	]
#resume continuous aggr01;
#select aggr01(); #should return 6
% .L2 # table_name
% L2 # name
% int # type
% 1 # length
[ 6	]
#select aggr01(); #should return 6
% .L2 # table_name
% L2 # name
% int # type
% 1 # length
[ 6	]
#stop continuous aggr01;
#drop function aggr01;
#drop table ftmp;

# 16:27:07 >  
# 16:27:07 >  "Done."
# 16:27:07 >  

